### PR TITLE
Fix two issues and add better error reporting

### PR DESCRIPTION
**Pull request by TheThing**
* Fix `router.param` support
* Add array of route support
* Generate better error if handler is not a function
* Made some optimization

Optimised instanceToWrap[method] so it should now be V8 optimized.

### CODE REVIEW

**Comment by TheThing** [comment]
This should fix both https://github.com/alex-whitney/express-promise-router/issues/21 and https://github.com/alex-whitney/express-promise-router/issues/12.

File: lib/express-promise-router.js [hunk 1]
```diff
@@ -8,6 +8,11 @@ var isPromise = function (obj) {
 };
 
 var wrapHandler = function (handler) {
+    if ('function' !== typeof handler) {
+        var type = Object.prototype.toString.call(handler);
+        var msg = 'Expected a callback function but got a ' + type;
+        throw new Error(msg);
+    }
```
Comment (alex-whitney) [inline lib/express-promise-router.js]: :+1:

File: package.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     "chai": "^1.9.1",
     "express": "4.x",
     "grunt": "^0.4.4",
+    "grunt-cli": "^0.1.13",
```
Comment (alex-whitney) [inline package.json]: Isn't this supposed to be installed globally, not locally for each project?

Comment (TheThing) [inline package.json]: I think grunt separated the cli commands from the main base project a while ago. In any case, it's good practice to have everything in the devDependancy that is required to run the tests and locally on my machine I don't have grunt-cli tools installed globally.

**Pull request merged**